rpc/tests/p2p.rs

```diff
@@ -0,0 +1,296 @@
+        true
+    );
+
+    client
+        .p2p_unblock_peer(&addr_info.id)
+        .await
+        .expect("failed to block peer");
+    rpc_call_delay().await;
+
+    assert_eq!(
+        client
+            .p2p_list_blocked_peers()
+            .await
+            .expect("failed to get blocked peer list")
+            .contains(&addr_info.id),
+        false
+    );
+
+    task.abort();
+}
+
+#[tokio::test]
+async fn bandwidth_stats_test() {
+    // just check whether we can get the data without error, node could have been running any
+    // amount of time, so any value should be valid.
+    let client = new_test_client(AuthLevel::Admin).await.unwrap();
+    client
+        .p2p_bandwidth_stats()
+        .await
+        .expect("failed to get bandwidth stats");
+}
+
+#[tokio::test]
+async fn bandwidth_for_peer_test() {
+    use libp2p::{identity, PeerId};
+
+    let local_key = identity::Keypair::generate_ed25519();
+    let local_peer_id = p2p::PeerId(PeerId::from(local_key.public()));
+
+    let client = new_test_client(AuthLevel::Admin).await.unwrap();
+    let stats = client
+        .p2p_bandwidth_for_peer(&local_peer_id)
+        .await
+        .expect("failed to get bandwidth stats for peer");
+
+    // where should be no data exchanged with peer that we're not connected to
+    assert_eq!(stats.total_in, 0.0);
+    assert_eq!(stats.total_out, 0.0);
+    assert_eq!(stats.rate_in, 0.0);
+    assert_eq!(stats.rate_out, 0.0);
+}
+
+#[tokio::test]
+async fn bandwidth_for_protocol_test() {
+    let client = new_test_client(AuthLevel::Admin).await.unwrap();
+
+    // query for nonsense protocol name so that we get all zeros in response
+    // until we have better way of inducing traffic
+    let stats = client
+        .p2p_bandwidth_for_protocol("/foo/bar")
+        .await
+        .expect("failed to get bandwidth stats");
+    assert_eq!(stats.total_in, 0.0);
+    assert_eq!(stats.total_out, 0.0);
+    assert_eq!(stats.rate_in, 0.0);
+    assert_eq!(stats.rate_out, 0.0);
+}
+
+#[tokio::test]
+async fn nat_status_test() {
+    // just query for status and make sure no errors happen, since any value is potentially correct
+    let client = new_test_client(AuthLevel::Admin).await.unwrap();
+    let _ = client
+        .p2p_nat_status()
+        .await
+        .expect("failed to query NAT status");
+}
+
+#[tokio::test]
+async fn peer_info_test() {
+    let (addr_info, task) = utils::tiny_node::start_tiny_node()
+        .await
+        .expect("failed to spin up second node");
+    log::debug!("{addr_info:?}");
+    let client = new_test_client(AuthLevel::Admin).await.unwrap();
+
+    client
+        .p2p_connect(&addr_info)
+        .await
+        .expect("request to connect to second node failed");
+    rpc_call_delay().await;
+
+    assert!(
+        client
+            .p2p_connectedness(&addr_info.id)
+            .await
+            .expect("failed to check connection to peer after connect request")
```

Comment:
save the result in a variable and then use `assert_eq` instead. check other places too

Reply:
changed